### PR TITLE
Jetpack Manage: Plugin management 'update plugin' link - show dialog, convert to button

### PR DESCRIPTION
**Pull request by coder-karen**
Related to https://github.com/Automattic/jetpack-manage/issues/116

## Proposed Changes

* This PR adds a Dialog component to the 'Update to xxx' link that shows on the plugin management page when a plugin has an update available.
* This uses the ConfirmModal component, and adds a new prop to pass the site count through (to UpdatePlugin). This seemed better than excluding the sitecount or a link to the sites with the plugin in question, and omitting that information means the modal is still sparse in information.
* The wording is similar to the dialog wording that shows if you use the bulk edit 'Update Plugin' button at the top right of the plugin management page. That uses a different component - see `/client/my-sites/plugins/hooks/use-get-dialog-text/types.d.ts` - but for flexibility and ease of use I've opted for the ConfirmModal.
* This PR also converts the link to a button, matching the other Update Plugin button in the table header.

## Testing Instructions

* Within the Jetpack Manage dashboard, visit the plugin management page: http://jetpack.cloud.localhost:3000/plugins/manage
* Find a plugin that needs updating on a Jetpack connected site that needs updating (download an older version of a plugin on a site for testing purposes if needed).
* Click on the 'Update to' button
* Notice the modal. The wording should say the name of plugin, and the number of sites needing that plugin updated. Click ok and the plugin should update.
* The same should be visible when visiting an individual sites plugin management page
* The tests should pass as well. They will pass in the PR, but can also be tested locally: `yarn run test-client client/my-sites/plugins/plugin-management-v2/test/update-plugin.test.tsx`

### Screenshots:
On the plugin management page:
Screenshot 2024-01-03 at 09 46 40
After clicking the 'Update to' button (showing the dialog):
Screenshot 2024-01-04 at 13 55 46

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/my-sites/plugins/plugin-management-v2/update-plugin/index.tsx [hunk 1]
```diff
@@ -43,6 +53,40 @@ export default function UpdatePlugin( { plugin, selectedSite, className, updateP
...
+	const pluginUpdateConfirmationTitle = translate( 'Update %(plugin)s', {
+		args: {
+			plugin: plugin.name ?? plugin.slug,
+		},
+	} );
+
+	const showConfirmationModal = () => (
+		<ConfirmModal
+			isVisible={ displayConfirmModal }
+			confirmButtonLabel={ translate( 'Yes' ) }
```
Comment (vitozev) [inline client/my-sites/plugins/plugin-management-v2/update-plugin/index.tsx]: Can we change the "Yes" to "Update" to match the wording of the bulk action update modal?

File: client/my-sites/plugins/plugin-management-v2/update-plugin/style.scss [hunk 2]
```diff
@@ -22,21 +22,12 @@
 }
 
 button.update-plugin__new-version {
-	text-decoration: underline;
-	font-weight: 600;
+	background-color: var(--studio-jetpack-green-50);
+	color: #fff;
```
Comment (vitozev) [inline client/my-sites/plugins/plugin-management-v2/update-plugin/style.scss]:
```suggestion
	color: var(--studio-white);
```

Nitpick: We could use the respective variable for the white color.

**Review by vitozev — COMMENTED** [review]
I tested the flows on both WordPress.com (as this code is being used there as well) and Jetpack Cloud, and everything is working as expected.

I've left a few minor comments.

File: client/my-sites/plugins/plugin-management-v2/update-plugin/index.tsx (hunk 1 above)
Comment (coder-karen) [reply client/my-sites/plugins/plugin-management-v2/update-plugin/index.tsx]: Definitely, added here: eee45373e353c2178148916e1da36857f1b31c8d

File: client/my-sites/plugins/plugin-management-v2/update-plugin/style.scss (hunk 2 above)
Comment (coder-karen) [reply client/my-sites/plugins/plugin-management-v2/update-plugin/style.scss]: Got it: 4500248e05836b4fb09bf8b93bc0e81dbdb92f7e

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/10670275

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @coder-karen for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/10670275).

**Pull request merged**